### ISSUE TITLE
Rails 4.2.3 regression on unscoped belongs_to

### ISSUE THREAD

**Issue by everton** [rails/rails#20721]
When including belongs_to relationships which cleans (unscope) default_scopes into new scopes, the default_scope remains applied.

Following script reproduces a regression introduced in 4.2.3.rc1

> https://gist.github.com/everton/7eed6c00c42712cc0a1f

**Obs.:** The problem occur only if the new scope use the "regular" association name, if the name of association does not reflect the name of the class it works perfeclty, what enable a workaround while this is not fixed:

```ruby
class Comment
  belongs_to :post, -> { unscope(where: :disabled_at) }
  belongs_to :post_even_if_deleted, -> { unscope(where: :disabled_at) }, 
    class_name: 'Post', foreign_key: :post_id

  scope :sorted_post_title, -> {
    # includes(:post).order('posts.title ASC')
    includes(:post_even_if_deleted).order('posts.title ASC')
  }
end
```

This issue is a response to @rafaelfranca request into #20679

**Comment by charlesvallieres** [comment]
Hi! Sorry to bring this back but we ran into this same exact issue and I found my way here.
I did some test against the 4-2-stable branch and it seems it still doesn't work.
@everton's gist is still failing for me, but maybe I am doing something wrong
(https://gist.github.com/charlesvallieres/17eb1d1d3a1248f19916 is out exact use case)

Thank you!

**Comment by adamnbowen** [comment]
I can confirm that this issue still exists—I've tested @everton's bug report file in 4.2.3, 4.2.4, 4.2.5.1, 4-2-stable, 5.0.0.beta2 and master, and it exists in all of them: only 4.2.2 works.

**Comment by everton** [comment]
Sorry, I forgot to notify it (because my code still working with workarounds and didn't checked the master) when this issue was closed, but Yes, it still broken, even on 5.0.0.beta2

**Comment by allaire** [comment]
Can we have a status update on this @senny @sgrif ?

Thanks a lot!

**Comment by senny** [comment]
@allaire I don't know any specifics about the fix. Let's wait for @sgrif

**Comment by everton** [comment]
5.1.2 it still broken. 

It seems to be fixed by #29611 which was merged into master (this was a 2 years birthday gift for the issue :gift: )
